pubsub/rabbitmq/rabbitmq.go

```diff
@@ -409,11 +450,14 @@ func (r *rabbitMQ) handleMessage(channel rabbitMQChannelBroker, d amqp.Delivery,
 	return err
 }
 
+// this function call should be wrapped by channelMutex.
```

Comment:
`ensureExchangeDeclared` may have potential concurrency issue. It should use channelMutex instead of wrapping the lock from `Publish` only, since it's called from multiple places.

Reply:
`ensureExchangeDeclared` is indeed called from multiple places, but it is always guarded by the mutex set in a function up the stack. This is the reason I added mutex in `Publish`